app/controllers/answers_controller.rb

```diff
@@ -24,12 +24,31 @@ def create
     end
   end
 
+  def update
+    @answer = Answer.find(params[:id])
+    @answer.update(answer_params)
+  end
+
   def destroy
-    @answer.destroy
-    redirect_to @question
-    flash[:notice] = 'Your answer deleted'
+    if @answer.user == current_user
+      @answer.destroy
+      flash[:notice] = 'Your answer deleted'
+    end
+  end
+
+  def mark_best
+    if @question.user_id == current_user.id
+      @answer.mark_best
+    else
+      raise "cant mark best"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.